File: server/models/book.js
const Joi = require('joi')
const mongoose = require('mongoose')

let bookSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        minlength: 1,
        maxlength: 255
    },
    subtitle: {
        type: String,
        required: true,
        minlength: 1,
        maxlength: 255
        
    },
    description: {
        type: String,
        required: true,
        minlength: 1,
        maxlength: 2000
    },
    author: {
        type: String,
        required: true,
        minlength: 1,
        maxlength: 255
    },
    link: String,
    coverLink: String
},
{timestamps:true})

const Book= mongoose.model('Book', bookSchema)

function validateBook (book) {
    const schema = {
        name: Joi.string().min(2).max(50).required(),
        subtitle: Joi.string().min(2).max(255).required(),
        description: Joi.string().min(2).max(2000).required(),
        author: Joi.string().min(2).max(255).required(),
    }

    return Joi.validate(book, schema)
}
function validateSearch (book) {
    const schema = {
        name: Joi.string().min(2).max(50),
        subtitle: Joi.string().min(2).max(255),
        description: Joi.string().min(2).max(255),
        author: Joi.string().min(2).max(255)
    }

    return Joi.validate(book, schema)
}
exports.Book = Book
exports.validate= validateBook
exports.validateSearch= validateSearch
//validation required

File: apps/student_app/lib/shared/models/book.dart
class Book {
  String id;
  String name;
  String subtitle;
  String description;
  String author;
  String link;
  String coverLink;

  Book(
      {this.id,
      this.name,
      this.subtitle,
      this.description,
      this.author,
      this.link,
      this.coverLink});

  Book.fromJSON(var json) {
    id = json['_id'];
    name = json['name'];
    subtitle = json['subtitle'];
    description = json['description'];
    author = json['author'];
    link = json['link'];
    coverLink = json['coverLink'];
  }

  Map<String, dynamic> toMap() {
    return {
      '_id': id,
      'name': name,
      'subtitle': subtitle,
      'description': description,
      'author': author,
      'link': link,
      'coverLink': coverLink
    };
  }

  @override
  String toString() {
    return 'Book{id: $id, name: $name, subtitle: $subtitle,description: $description,author: $author,link: $link,coverLink: $coverLink}';
  }
}
